ios/FluentUI/BottomSheet/BottomSheetViewController.swift

```diff
@@ -0,0 +1,404 @@
+        if animator?.isRunning ?? false {
+           return
+        }
+
+        setupAnimator()
+    }
+
+    private func panningStop(in verticalPos: CGFloat, velocity: CGFloat) {
+        panGestureRecognizer.isEnabled = false
+
+        let windowHeight = view.window?.frame.height ?? 0
+        let shouldReverse: Bool
+        let futureState: BottomSheetViewState
+        switch currentState {
+        case .expand:
+            if verticalPos >= windowHeight / 2 || velocity > Constants.velocityThreshold {
+                shouldReverse = false
+                futureState = .collapse
+            } else {
+                shouldReverse = true
+                futureState = .expand
+            }
+        case .collapse:
+            if verticalPos <= -windowHeight / 2 || velocity <= -1 * Constants.velocityThreshold {
+                shouldReverse = false
+                futureState = .expand
+            } else {
+                shouldReverse = true
+                futureState = .collapse
+            }
+        }
+
+        animator?.isReversed = shouldReverse
+        addAnimatorCompletion(to: futureState, completion: nil)
+        animator?.continueAnimation(withTimingParameters: nil, durationFactor: 0)
+    }
+
+    private func panningProgress(in verticalPos: CGFloat) {
+        if animator?.isRunning ?? false {
+           return
+        }
+
+        let windowHeight = view.window?.frame.height ?? 0
+        var progress = -1 * verticalPos / (windowHeight - collapsedHeight)
+
+        if currentState == .expand {
+            progress *= -1
+        }
+
+        animator?.fractionComplete = progress
+    }
+
+// MARK: - Resize Handle utilities
+
+    @objc private func handleResizingHandleViewTap(gesture: UITapGestureRecognizer) {
+        if animator?.isRunning ?? false {
+            return
+        }
+
+        if gesture.state == .recognized {
+            let futureState: BottomSheetViewState
+            switch currentState {
+            case .collapse:
+                futureState = .expand
+            case .expand:
+                futureState = .collapse
+            }
+
+            setupAnimator()
+            addAnimatorCompletion(to: futureState)
+            animator?.startAnimation()
+        }
+    }
+
+    private func updateResizingHandleViewAccessibility() {
+        if currentState == .expand {
+            resizingHandleView.accessibilityLabel = "Accessibility.Drawer.ResizingHandle.Label.Collapse".localized
+            resizingHandleView.accessibilityHint = "Accessibility.Drawer.ResizingHandle.Hint.Collapse".localized
+        } else {
+            resizingHandleView.accessibilityLabel = "Accessibility.Drawer.ResizingHandle.Label.Expand".localized
+            resizingHandleView.accessibilityHint = "Accessibility.Drawer.ResizingHandle.Hint.Expand".localized
+        }
+    }
+
+// MARK: - Dimming View utilities
+    private func setupDimmingView() {
+        if let rootView = view.superview {
```

Comment:
I wonder if there's a possible architecture where we move everything one level down, so self.view is transparent and the bottom sheet + the dimming view are the sibling children of self.view. 
Injecting into the superview sort of imposes the same constraints on the client (view.superview should cover the full app screen because everything should dim), but we're not as explicit about it. Would also get rid of the responsibility to clean up the dimming view.
@anandrajeswaran thoughts?

Reply:
i think the problem here is that as soon as we try to make self.view transparent, we run into the same accessibility issues as before.

Reply:
@harrieshin is that what causes the accessibility issues? I thought it had to do with the view controllers being child view controllers as opposed to sibling view controllers, but am likely not remembering correctly

Reply:
Never mind. @lcapkovic corrected me offline. The accessibility issues I remember were caused by using presentViewController, which accessibility interprets to mean at least a semi-modal presentation. Curious to hear more about what kind of issues you saw when making self.view transparent (i.e. a default view that simply contains other views)